cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueConsumer.java

```diff
@@ -133,26 +119,37 @@ public void close() throws IOException {
     if (closed) {
       return;
     }
-    try {
-      stateStore.saveState(new HBaseConsumerState(startRow, getConfig().getGroupId(), getConfig().getInstanceId()));
-    } finally {
-      Closeables.closeQuietly(queueStrategy);
-      hTable.close();
-      closed = true;
-    }
+    closed = true;
+    Closeables.closeQuietly(queueStrategy);
+    Closeables.closeQuietly(stateStore);
+    Closeables.closeQuietly(hTable);
+  }
+
+  @Override
+  public void startTx(Transaction tx) {
+    super.startTx(tx);
+    stateStore.startTx(tx);
+  }
+
+  @Override
+  public boolean rollbackTx() throws Exception {
+    return super.rollbackTx() && stateStore.rollbackTx();
```

Comment:
why && ? Don't you want to rollback as much as possible even if the super class fails to rollback?